osdep/mac/clipboard.swift

```diff
@@ -73,4 +74,218 @@ class Clipboard: NSObject {
+                }
+            case PL_COLOR_PRIM_DCI_P3:
+                if plcsp.transfer == PL_COLOR_TRC_BT_1886 {
+                    cgSpaceName = CGColorSpace.dcip3
+                }
+            case PL_COLOR_PRIM_BT_2020:
+                if plcsp.transfer == PL_COLOR_TRC_BT_1886 {
+                    cgSpaceName = CGColorSpace.itur_2020
+                }
+            case PL_COLOR_PRIM_ADOBE:
+                cgSpaceName = CGColorSpace.adobeRGB1998
+            case PL_COLOR_PRIM_APPLE:
+                if plcsp.transfer == PL_COLOR_TRC_LINEAR {
+                    cgSpaceName = CGColorSpace.genericRGBLinear
+                }
+            default:
+                break
+            }
+        }
+
+        if cgSpaceName == nil {
+            compatible = false
+            cgSpaceName = CGColorSpace.sRGB
+            plcsp.primaries = PL_COLOR_PRIM_BT_709
+            plcsp.transfer = PL_COLOR_TRC_SRGB
+        }
+
+        var nscsp: NSColorSpace?
+
+        if !gray && image.icc_profile != nil {
+            nscsp = NSColorSpace(iccProfileData: Data(bytes: image.icc_profile.pointee.data, count: image.icc_profile.pointee.size))
+        } else if let spaceName = cgSpaceName {
+            guard let cgSpace = CGColorSpace(name: spaceName) else {
+                return CLIPBOARD_FAILED
+            }
+            nscsp = NSColorSpace(cgColorSpace: cgSpace)
+        }
+
+        if let csp = nscsp {
+            guard let retagged = rep.retagging(with: csp) else {
+                return CLIPBOARD_FAILED
+            }
+
+            rep = retagged
+        }
+
+        if !compatible {
+            var dest = mp_image()
+            mp_image_setfmt(&dest, Int32(imgfmt.rawValue))
+            mp_image_set_size(&dest, image.w, image.h)
+            rep.getBitmapDataPlanes(planes)
+
+            withUnsafeMutableBytes(of: &dest.stride) { (stridePtr) in
+                withUnsafeMutableBytes(of: &dest.planes) { (planesPtr) in
+                    let destStrides = stridePtr.baseAddress!.assumingMemoryBound(to: type(of: image.stride.0))
+                    let destPlanes = planesPtr.baseAddress!.assumingMemoryBound(to: type(of: image.planes.0))
+
+                    for i in 0..<Int(MP_MAX_PLANES) {
+                        destPlanes[i] = planes[i]
+                        destStrides[i] = bytesPerRow
+                    }
+                }
+            }
+
+            dest.params.repr.sys = PL_COLOR_SYSTEM_RGB
+            dest.params.repr.levels = PL_COLOR_LEVELS_FULL
+            dest.params.repr.alpha = rep.hasAlpha ? ((rep.bitmapFormat.contains(.alphaNonpremultiplied)) ? PL_ALPHA_INDEPENDENT : PL_ALPHA_PREMULTIPLIED) : PL_ALPHA_UNKNOWN
+            dest.params.repr.bits.sample_depth = Int32(bps)
+            dest.params.repr.bits.color_depth = Int32(bps)
+
+            dest.params.color = plcsp
+
+            if mp_image_swscale(&dest, &image, SWS_FULL_CHR_H_INT | SWS_FULL_CHR_H_INP | SWS_ACCURATE_RND) < 0 {
```

Comment:
This should use the same pattern and options as other screenshot code. See https://github.com/mpv-player/mpv/commit/2c43d2b75a88b8e0e8f0a715f993ffc1c8977d13

I will remove  `mp_image_swscale` to avoid confusion.

Reply:
Why add more brittle boilerplate? Seems more reasonable to me to add `global` and `log` options to `mp_image_swscale` and then remove the duplication in `screenshot.c`, `image_writer.c`, etc.

Reply:
Rebased with this change; the relevant changeset is net-red (+11 -20) so I think it's a reasonable reduction in duplication. Also improved one path through player/screenshot that wasn't using the global options (nor checking for errors).